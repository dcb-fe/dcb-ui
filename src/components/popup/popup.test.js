import Popup from './popup.vue';
import { shallowMount } from '@vue/test-utils';

describe('Popup', () => {
  describe('视图', () => {
    test('visible 为 false 时不渲染遮罩和内容', () => {
      const wrapper = shallowMount(Popup, {
        slots: {
          default: '内容',
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });

    test('visible 为 true 时渲染遮罩和内容', () => {
      const wrapper = shallowMount(Popup, {
        propsData: {
          visible: true,
        },
        slots: {
          default: '内容',
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('逻辑', () => {
    test('初始 visible 为 false 时不触发 close 事件', () => {
      const handleClose = jest.fn();
      shallowMount(Popup, {
        slots: {
          default: '内容',
        },
        listeners: {
          close: handleClose,
        },
      });
      expect(handleClose).not.toBeCalled();
    });

    test('初始 visible 为 true 时触发 open 事件', () => {
      const handleOpen = jest.fn();
      shallowMount(Popup, {
        propsData: {
          visible: true,
        },
        slots: {
          default: '内容',
        },
        listeners: {
          open: handleOpen,
        },
      });
      expect(handleOpen).toBeCalled().toBeCalledTimes(1);
    });
  });
});
