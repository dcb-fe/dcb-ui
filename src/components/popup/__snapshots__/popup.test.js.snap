// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup 视图 visible 为 false 时不渲染遮罩和内容 1`] = `
<portal-stub selector="d-portal-container" tag="DIV">
  <div class="d-popup d-popup-center d-popup-hidden" style="z-index: 5000;">
    <transition-stub name="d-transition-fade" appear="true">
      <!---->
    </transition-stub>
    <transition-stub name="d-transition-fade" appear="true">
      <!---->
    </transition-stub>
  </div>
</portal-stub>
`;

exports[`Popup 视图 visible 为 true 时渲染遮罩和内容 1`] = `
<portal-stub selector="d-portal-container" tag="DIV">
  <div class="d-popup d-popup-center" style="z-index: 5001;">
    <transition-stub name="d-transition-fade" appear="true">
      <div class="d-popup-mask"></div>
    </transition-stub>
    <transition-stub name="d-transition-fade" appear="true">内容</transition-stub>
  </div>
</portal-stub>
`;
